fe/fe-core/src/main/java/org/apache/doris/analysis/FunctionCallExpr.java

```diff
@@ -1337,7 +1337,8 @@ public void analyzeImpl(Analyzer analyzer) throws AnalysisException {
                     Function.CompareMode.IS_NONSTRICT_SUPERTYPE_OF);
         } else if (fnName.getFunction().equalsIgnoreCase("if")) {
             Type[] childTypes = collectChildReturnTypes();
-            Type assignmentCompatibleType = ScalarType.getAssignmentCompatibleType(childTypes[1], childTypes[2], true);
+            Type assignmentCompatibleType = Type.convertDateLikeTypeToV2(
+                    ScalarType.getAssignmentCompatibleType(childTypes[1], childTypes[2], true));
```

Comment:
because the assignmentCompatibleType may be date and childType is [date, null_type], and the null_type will be replaced to datev2 type, so we add this to ensure the assignmentCompatibleType is datev2 type